### PR TITLE
Add `custom_get_name` config option

### PR DESCRIPTION
**Pull request by eupharis**
Attached is what I'm thinking for a way to allow individual users of the library to pass their own `get_name` method.

In case they want to have generated python class types like `SubresourceUri` instead of just the default `.title()` version of `Subresourceuri`

https://github.com/sbrunner/jsonschema-gentypes/issues/843

NOTE: we'll need to swap `get_name` to `self.get_name` on all subclasses of `API` before we can merge this repo. I only did `API` and `APIv4` so far.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -7,3 +7,4 @@ __pycache__
 /dist
 /.github/changelog-generator-cache
 /tests/recursion.py
+/.vscode
```
Comment (sbrunner) [inline .gitignore:10]: This should be on your own `~/.gitignore` file...

File: tests/test_custom_get_name.py [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+            Union[
+                jsonschema_draft_04.JSONSchemaD4,
+                jsonschema_draft_2019_09_meta_data.JSONSchemaItemD2019,
+            ]
+        ],
+        Optional[str],
+        bool,
+    ],
+    str,
+]
```
Comment (sbrunner) [inline tests/test_custom_get_name.py:19]: Why this copy?

File: jsonschema_gentypes/configuration.py [hunk 3]
```diff
@@ -2,10 +2,12 @@
 Automatically generated file from a JSON schema.
 """
 
-from typing import Any, Literal, TypedDict, Union
+from typing import Any, Callable, Literal, Optional, TypedDict, Union
```
Comment (sbrunner) [inline jsonschema_gentypes/configuration.py:5]: This file is generated from `jsonschema_gentypes/schema.json` and describes the schema of the configurations file.

Then if we want to be able to configure it only with python code we should get the function definition out of this file :-)

But in fact, I didn't see well how you plan to use it?
Like in the test?

Comment (eupharis) [reply jsonschema_gentypes/configuration.py:5]: @sbrunner ok! I fixed the two other issues you pointed out.

I can totally get the definition out and instead put it in the `configuration` file.

what's the command to generate the schema.json from the configuration file?

I plan to use this exactly how I use it in the test.

Without the `custom_get_name`, you wind up with the `.title` version like `Subresourceuri` as the name and the test fails. I need the only-first-letter-capitalized version like `SubresourceUri`. 

I also invoke this by importing the `process_config` function from a Python script, not from the cli.

Comment (eupharis) [reply jsonschema_gentypes/configuration.py:5]: @sbrunner want me to pull that definition out of the `configuration` file? any other changes?

Comment (sbrunner) [reply jsonschema_gentypes/configuration.py:5]: > what's the command to generate the schema.json from the configuration file?

It's the inverse, this file is generated from this schema https://github.com/sbrunner/jsonschema-gentypes/blob/7c7fe0d6f5539c13334580762bfacef292c6edba/jsonschema_gentypes/schema.json

It's done by pre-commit :-)

**Comment by sbrunner** [comment]
That looks good to me :-)

**Comment by eupharis** [comment]
@sbrunner awesome! could we release it as new version?

The one failing check is this:

"requests.exceptions.HTTPError: 401 Client Error: Unauthorized for url: https://api.github.com/repos/sbrunner/jsonschema-gentypes/pulls/889/commits"

I think that's a spurious error maybe?

**Comment by sbrunner** [comment]
Released in version 2.6.0

**Comment by eupharis** [comment]
@sbrunner woohoo! thanks so much!

**Pull request merged**